examples/generate-csr.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This script generates a simple SAN CSR to be used with Let's Encrypt
+# CA. Mostly intended for "auth --csr" testing, but, since it's easily
+# auditable, feel free to adjust it and use it on your production web
+# server.
+import os
+import argparse
+from OpenSSL import crypto
+
+try:
+    from cryptography.hazmat.backends import default_backend
+    from cryptography.hazmat.primitives.asymmetric import rsa, ec
+    from cryptography.hazmat.primitives.serialization import Encoding, PrivateFormat, NoEncryption
+    cryptography=True
+except ImportError:
+    cryptography=False
+
+CWD = os.getcwd()
+
+
+def generate_parser():
+    one_parser = argparse.ArgumentParser(description='Generate a simple SAN CSR')
+    one_parser.add_argument('domain', nargs='+', help='A FQDN to add in the CSR.')
+    one_parser.add_argument('--key-path', default=os.path.join(CWD, 'key.pem'),
+                            help='Path for the generated key '
+                                 '(default is key.pem in current directory)')
+    one_parser.add_argument('--csr-path', default=os.path.join(CWD, 'csr.der'),
+                            help='Path for the generated certificate '
+                                 '(default is csr.der in current directory)')
+    one_parser.add_argument('--key-type', default='RSA',
+                            choices=['RSA', 'ECDSA'],
+                            help='Key type to use, ECDSA is supported only '
+                                 'if cryptography module is available')
+
+    return one_parser
+
+
+def main(domains, key_path, csr_path, key_type):
+    san = ', '.join(['DNS: {0}'.format(item) for item in domains])
```

Comment:
There's no spaces in the original, not sure there should be here

Reply:
Done.